trade-finance-manager-ui/server/controllers/helpers/get-success-banner-message.helper.ts

```diff
@@ -0,0 +1,53 @@
+import { Request } from 'express';
+import { TFM_DEAL_CANCELLATION_STATUS, TfmDealCancellationWithStatus, DATE_FORMATS, Deal, getUkefDealId } from '@ukef/dtfs2-common';
+import { format } from 'date-fns';
+import api from '../../api';
+import { isDealCancellationEnabled } from './deal-cancellation-enabled.helper';
+import { getFlashSuccessMessage } from '../../helpers/get-flash-success-message';
+
+/**
+ * Returns deal cancellation message if the deal is scheduled for cancellation
+ * @param dealSnapshot - the deal
+ * @param userToken - the user token
+ * @returns the success message to be shown or null
+ */
+export const getScheduledCancellationBannerMessage = async ({ dealSnapshot, userToken }: { dealSnapshot: Deal; userToken: string }): Promise<string | null> => {
+  const { submissionType, _id: dealId } = dealSnapshot;
+  const ukefDealId = getUkefDealId(dealSnapshot);
+
+  const dealCancellationIsEnabled = isDealCancellationEnabled(submissionType);
+
+  if (!dealCancellationIsEnabled) {
+    return null;
+  }
+
+  const cancellation = (await api.getDealCancellation(dealId.toString(), userToken)) as TfmDealCancellationWithStatus;
+
+  const dealIsScheduledToBeCancelled = cancellation.status === TFM_DEAL_CANCELLATION_STATUS.SCHEDULED && cancellation.effectiveFrom;
+
+  if (!dealIsScheduledToBeCancelled) {
+    return null;
+  }
+
+  const formattedDate = format(new Date(cancellation.effectiveFrom), DATE_FORMATS.D_MMMM_YYYY);
+
+  return `Deal ${ukefDealId} scheduled for cancellation on ${formattedDate}`;
+};
+
+/**
+ * Determines if a success message should be shown and returns the highest priority message.
+ * @param params - the deal
+ * @param params.dealSnapshot - the deal
+ * @param params.userToken - the user token
+ * @param params.req the express request object
+ * @returns the success message to be shown or null
```

Comment:
Does this ever return null? It depends on the behaviour of `getFlashSuccessMessage` on an empty array...